examples/nidcpower_source_dc_voltage/measurement.py

```diff
@@ -77,49 +77,58 @@ def measure(
             # returns a context manager that aborts the measurement when the
             # function returns or raises an exception.
-            with channels.initiate():
-                # Wait for the outputs to settle.
-                timeout = source_delay + 10.0
-                _wait_for_event(
-                    channels, cancellation_event, nidcpower.enums.Event.SOURCE_COMPLETE, timeout
-                )
-
-                # Measure the voltage and current for each output.
-                measurements: List[_Measurement] = channels.measure_multiple()
-
-                # Determine whether the outputs are in compliance.
-                for index, connection in enumerate(connections):
-                    channel = connection.session.channels[connection.channel_name]
-                    in_compliance = channel.query_in_compliance()
-                    measurements[index] = measurements[index]._replace(in_compliance=in_compliance)
-
-    _log_measurements(connections, measurements)
+            with ExitStack() as stack:
+                for session_info in session_infos:
+                    channels = session_info.session.channels[session_info.channel_list]
+                    stack.enter_context(channels.initiate())
+
+                for session_info in session_infos:
+                    channels = session_info.session.channels[session_info.channel_list]
+                    timeout = source_delay + 10.0
+                    _wait_for_event(
+                        channels, cancellation_event, nidcpower.enums.Event.SOURCE_COMPLETE, timeout
+                    )
+
+                measurements: List[_Measurement] = []
+                measured_sites, measured_pins = [], []
+                for session_info in session_infos:
+                    channels = session_info.session.channels[session_info.channel_list]
+                    session_measurements: List[_Measurement] = channels.measure_multiple()
+
+                    for measurement, channel_mapping in zip(
+                        session_measurements, session_info.channel_mappings
+                    ):
+                        measured_pins.append(channel_mapping.pin_or_relay_name)
+                        measured_sites.append(channel_mapping.site)
+                        in_compliance = session_info.session.channels[
+                            channel_mapping.channel
+                        ].query_in_compliance()
+                        measurement._replace(in_compliance=in_compliance)
+
+                    measurements.extend(session_measurements)
+
+    _log_measurements(measured_pins, measured_sites, measurements)
     logging.info("Completed measurement")
     return (
-        [connection.site for connection in connections],
-        [connection.pin_or_relay_name for connection in connections],
+        [site for site in measured_sites],
+        [pin for pin in measured_pins],
```

Comment:
measured_sites and measured_pins are already lists, so there is no need for a list comprehension.
```suggestion
        measured_sites,
        measured_pins,
```

Reply:
Thank you for pointing it out. I've removed the list comprehension.